src/Runner/Runner.php

```diff
@@ -89,26 +119,245 @@ public function __construct(
         $this->cacheManager = $cacheManager;
         $this->directory = $directory ?? new Directory('');
         $this->stopOnViolation = $stopOnViolation;
+        $this->parallelConfig = $parallelConfig ?? ParallelConfig::sequential();
+        $this->input = $input;
+        $this->configFile = $configFile;
     }
 
     /**
-     * @return array<string, array{appliedFixers: list<string>, diff: string}>
+     * @param iterable<\SplFileInfo> $fileIterator
+     */
+    public function setFileIterator(iterable $fileIterator): void
+    {
+        $this->fileIterator = $fileIterator;
+    }
+
+    /**
+     * @return _RunResult
      */
     public function fix(): array
+    {
+        return $this->parallelConfig->getMaxProcesses() > 1 && null !== $this->input
```

Comment:
Because for backward compatibility reasons input is added to `Runner`'s constructor as an optional argument with default `null` value, and at the same time it's required for `ProcessFactory` use in `fixParallel()`. So effectively parallel runner is available only if input was passed to the runner's instance.

Reply:
ack. please mark the parameter with `@todo` and indicate that it should stop being optional for next MAJOR release, so we don't forget to clean this up